Dockerfile

```diff
@@ -1,13 +1,17 @@
-FROM python:3.6-alpine
+FROM pypy:3.9-slim-buster
 
-ARG IRRD_VERSION=4.0.8
+ARG IRRD_VERSION=4.2.8
 
-RUN apk update \
- && apk add --no-cache --virtual .build-deps \
-        build-base musl-dev gcc python3 python3-dev postgresql-dev \
- && pip install --no-cache-dir cython irrd==$IRRD_VERSION \
- && rm -rf /var/cache/irrd \
- && apk del --no-cache .build-deps \
- && apk add gnupg postgresql
+COPY run.sh /opt/irrd/run
 
-CMD ["/usr/local/bin/twistd -n irrd"]
+RUN apt-get update \
+ && apt-get install -y build-essential gcc gnupg libpq-dev dumb-init \
+ && pip install --no-cache-dir -U pip \
+ && pip install --no-cache-dir irrd==$IRRD_VERSION \
+ && apt-get clean autoclean \
+ && apt-get autoremove -y \
+ && rm -rf /var/lib/apt/lists/*
+
+ENTRYPOINT ["/usr/bin/dumb-init", "--"]
+
+CMD ["/opt/irrd/run"]
```

Comment:
The reason I asked about the way you created the image is that I wanted to maintain the irrd image, given that maintainers do not consider doing it.

I am still using alpine, but in a multistage build fashion -- https://github.com/srl-labs/irrd-container/blob/main/Dockerfile

Now, this image doesn't have the necessary steps you keep at run.sh...

Reply:
Yeah, so just like above, a good part of the run script is around GPG keyring that some people may want to use (otherwise, it'll show some benign warnings without affecting the start up). Or, in other words, it's optional.
Other things refer to the user and PID file, or the `irrd_database_upgrade` that'll be executed when bumping the IRRd version.
Another thing is replacing the replacing the database URL with a value collected from the environment (since it contains sensitive data, it's a common practice in Kubernetes environments but not limited to to have the credentials passed in as env vars).
Regarding Alpine, since a few years ago, for any Python project, I'm staying away, mostly because most of the Python packages don't publish a wheel for Alpine. See also: https://pythonspeed.com/articles/alpine-docker-python/ and https://rpep.dev/posts/alpine-python-antipattern/.